Add recipe for bazel-mode

codesuki opened the pull request:
### Brief summary of what the package does

Supports basic syntax highlighting and formatting for Bazel files.

### Direct link to the package repository

https://github.com/codesuki/bazel-mode

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [ ] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

codesuki commented:
`C-c C-c` in the recipe shows me the following error.
```
Install new package? (y or n) y
package-version-join: Wrong type argument: arrayp, nil
```

```
;; Please check the following package descriptor.
;; If the correct package description or dependencies are missing,
;; then the source .el file is likely malformed, and should be fixed.
(bazel-mode .
            [(20171221 1514)
             ((emacs
               (24 3)))
             "A major mode for editing Bazel files" single
             ((:commit . "bf41f73c0df7a414c55fc9ac07573300f3161258")
              (:keywords "languages" "bazel")
              (:authors
               ("Neri Marschik"))
              (:maintainer "Neri Marschik")
              (:url . "https://github.com/codesuki/bazel-mode"))])
```

purcell commented:
Thanks. The error you posted looks odd, sorry, but the build was happy. Your pasted output includes a dependency on Emacs 24.3, which is required by your code, but that package dependency is missing in the latest `master` version of the code, so I'm not sure what's going on there. Everything looks fine apart from that.

codesuki commented:
I removed that since it was a leftover from code I removed before. So I tried if it works if I simplify the version scheme to be similar to my other Emacs packages. 

Anyhow, I tried to build other recipes locally and they all gave me the same message. So I think it's something wrong with my emacs. 

If it works on the ci it's probably ok.

purcell commented:
> I removed that since it was a leftover from code I removed before. So I tried if it works if I simplify the version scheme to be similar to my other Emacs packages.
> 
> Anyhow, I tried to build other recipes locally and they all gave me the same message. So I think it's something wrong with my emacs.
> 
> If it works on the ci it's probably ok.

Sorry, perhaps I wasn't clear. Your code uses `setq-local`, so you _must_ declare a dependency on `(emacs "24.3")` in your `Package-Requires` header. The CI doesn't check for that.

codesuki commented:
Oh sorry. Didn't get that. I'll change it back.

codesuki commented:
I added back the dependency.

purcell commented:
Great, thanks - merged!

codesuki commented:
Thanks!

odanoburu commented:
@codesuki I'm having this same problem you (are? were?) having:
```
package-version-join: Wrong type argument: arrayp, nil
```
did you solve it eventually, or did you guys just merge because CI showed everything was fine?

thanks for the info!

codesuki commented:
We just merged and it seems to have worked out nicely. I am pretty sure it has something to do with my local emacs installation that I cannot get that to work.
